akka-persistence-typed/src/main/resources/reference.conf

```diff
@@ -20,4 +20,25 @@ akka.persistence.typed {
   # this may happen while it receives commands while it is recovering events or while it is persisting events
   log-stashing = off
 
+  # Describes the algorithms and behavior of retention within
+  # the snapshots and events comprising them.
+  retention {
+
+    # Snapshots are used to reduce playback/recovery times. This defines
+    # when a new snapshot is persisted. Defaults to a count algorithm,
+    # with 1000 messages, such that on 1001, the next snapshot is taken,
+    # and if successful, the events comprising the snapshot
+    # are deleted to free up space.
+    snapshot-size = 1000
+
+    # After a snapshot is successfully completed,
+    # - if 2: retain last maximum 2 *`snapshot-size` events and 3 snapshots (2 old + latest snapshot)
+    # - if 0: all events with equal or lower sequence number will no longer be retained
+    # Defaults to a count of 2 snapshots.
+    total-snapshots = 2
+
+    # Opt-in ability to not delete events, only delete snapshots.
+    # Defaults to disabled.
+    retain-snapshots-only = off
+  }
```

Comment:
Should it really even have config file based enabling?

Reply:
I don't think we should have config for this in Persistence because one would typically like to have different settings for different types of actors. Then we would have to put it into the journal config section and it becomes more complicated than what it is worth. Better with one programatic setting like what we already have for `snapshotEvery` in `EventSourcedBehavior`.

Reply:
Ah, yes that makes more sense. Changing..

Reply:
My original thought around this is how cluster sharding does it, and with config.